0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+An **accounting system** tracks a participant's accounts and the net balance owed between peers.
+
+A **settlement** is a reconciliation of a liability.
+
+A **settlement system** is a medium for exchanging value.
+
+## Overview
+
+This specification codifies a **settlement engine**, a common interface between an accounting system and heterogeneous settlement systems.
+
+Compatible settlement engine implementations enable peers to mutually reconcile one's liabilities through integrating with a shared settlement system.
+
+The purpose of a settlement engine is to perform two primary operations:
+
+1. Execute outgoing settlements to a peer
+2. Acknowledge incoming settlements from a peer
+
+Settlements may differ depending on the settlement engine implementation, such as transferring value on the underlying system or ledger, and/or using a custom protocol through which the peers mutually agree to reconcile a liability.
+
+The core test case for a settlement engine implementation is **the sum of amounts an instance is instructed to settle should eventually equal the sum of amounts the peer's instance acknowledges receipt of.**
+
+Varying implementations, arrangements, and conditions (such as connectivity to the ledger) may delay consistency of instructed and acknowledged settlements by an indeterminate amount of time.
+
+## Example Settlement Arrangements
+
+### Simple value transfers
+
+Suppose two peers, Alice and Bob, are operating the same or compatible settlement engine implementations, which settle using a ledger which both Alice and Bob maintain accounts on.
+
+1. Alice's accounting system instructs her settlement engine to perform a settlement of 5 units to Bob.
+2. Subsequently, Alice's settlement engine sends a value transfer of 5 units from Alice's ledger account to Bob's ledger account.
+   - To know where to send the payment, Alice may have configured her settlement engine with Bob's ledger identifier, or the settlement engine may automatically send a message to Bob's settlement engine to request Bob's ledger identifier.
+   - The settlement engine implementation should ensure that an incoming payment is only credited to the peer's account that sent it.
+3. Bob's settlement engine, always watching the ledger, recognizes the payment, correlates that Alice sent it, and notifies its accounting system of the incoming settlement for 5 units.
+
+### Payment channels
+
+Payment channels are special bilateral or multilateral settlement systems backed by escrows on an underlying shared ledger. They enable peers to adjust the allocation of these escrows among themselves. A participant can sign claims—messages with digital signatures entitling its peer to some greater portion of the escrow—which the peer can validate immediately without informing the underlying ledger. When the peers are done transacting, the most recent of many such claims is enforceable by the underlying ledger so the escrowed funds can be released and distributed accordingly.
```

Comment:
It might make sense to link to some external explanation of payment channels. There's a lot to go over here and having to explain how payment channels work in addition to how they integrate with a settlement engine makes this section kinda hard to read

Reply:
Good idea, it gets a bit unwieldy. (Makes me think, what if we had a separate RFC just devoted to using Interledger with payment channels!?)

Reply:
yeah, I think we definitely spend enough time explaining it in our various RFCs to make it worthwhile

Reply:
Removed all examples per comment here: https://github.com/interledger/rfcs/pull/536#discussion_r310256913

(But looked at the ILPv4 RFC, and the paychan explanation is pretty good; maybe I should link to that?)